Source/WebCore/Modules/WebGPU/Implementation/WebGPUDeviceImpl.cpp

```diff
@@ -187,34 +187,32 @@ Ref<BindGroupLayout> DeviceImpl::createBindGroupLayout(const BindGroupLayoutDesc
         };
     });
 
-    Vector<WGPUBindGroupLayoutEntry> backingEntries;
-    backingEntries.reserveInitialCapacity(descriptor.entries.size());
-    for (size_t i = 0; i < descriptor.entries.size(); ++i) {
-        const auto& entry = descriptor.entries[i];
-        backingEntries.uncheckedAppend({
+    Vector<WGPUBindGroupLayoutEntry> backingEntries(descriptor.entries.size(), [&](size_t i) {
+        auto& entry = descriptor.entries[i];
+        return WGPUBindGroupLayoutEntry {
             entry.externalTexture ? &backingExternalTextureEntries[i].chain : nullptr,
-            entry.binding,
-            m_convertToBackingContext->convertShaderStageFlagsToBacking(entry.visibility), {
-                nullptr,
-                entry.buffer ? m_convertToBackingContext->convertToBacking(entry.buffer->type) : WGPUBufferBindingType_Undefined,
-                entry.buffer ? entry.buffer->hasDynamicOffset : false,
-                entry.buffer ? entry.buffer->minBindingSize : 0,
-            }, {
-                nullptr,
-                entry.sampler ? m_convertToBackingContext->convertToBacking(entry.sampler->type) : WGPUSamplerBindingType_Undefined,
-            }, {
-                nullptr,
-                entry.texture ? m_convertToBackingContext->convertToBacking(entry.texture->sampleType) : WGPUTextureSampleType_Undefined,
-                entry.texture ? m_convertToBackingContext->convertToBacking(entry.texture->viewDimension) : WGPUTextureViewDimension_Undefined,
-                entry.texture ? entry.texture->multisampled : false,
-            }, {
-                nullptr,
-                entry.storageTexture ? m_convertToBackingContext->convertToBacking(entry.storageTexture->access) : WGPUStorageTextureAccess_Undefined,
-                entry.storageTexture ? m_convertToBackingContext->convertToBacking(entry.storageTexture->format) : WGPUTextureFormat_Undefined,
-                entry.storageTexture ? m_convertToBackingContext->convertToBacking(entry.storageTexture->viewDimension) : WGPUTextureViewDimension_Undefined,
-            },
-        });
-    }
+                entry.binding,
+                m_convertToBackingContext->convertShaderStageFlagsToBacking(entry.visibility), {
+                    nullptr,
+                    entry.buffer ? m_convertToBackingContext->convertToBacking(entry.buffer->type) : WGPUBufferBindingType_Undefined,
```

Comment:
Not your code, but all these ternary operations look funny. I wonder if the constructor could have been templated on entry.buffer fullness.